web/packages/teleport/src/User/UserContext.tsx

```diff
@@ -83,23 +82,24 @@ export function UserContextProvider(props: PropsWithChildren<unknown>) {
     return prefs.pinnedResources;
   }, []);
 
-  const updateClusterPinnedResources = useCallback(
-    async (clusterId: string, pinnedResources: string[]) => {
-      let currentPrefs = { ...clusterPreferences[clusterId] };
-      if (currentPrefs) {
-        currentPrefs.pinnedResources = pinnedResources;
-      } else {
-        currentPrefs = { pinnedResources };
-      }
-      clusterPreferences.current[clusterId] = currentPrefs;
-
-      return service.updateUserClusterPreferences(clusterId, {
-        ...preferences,
-        clusterPreferences: currentPrefs,
-      });
-    },
-    [preferences]
-  );
+  const updateClusterPinnedResources = async (
+    clusterId: string,
+    pinnedResources: string[]
+  ) => {
+    await setTimeout(() => {}, 1000);
+    let currentPrefs = { ...clusterPreferences[clusterId] };
```

Comment:
I realized that `{...undefined}` returns `{}` so ` if (currentPrefs)` always returns `true`.
Since we are using `useRef` we can just mutate the value:

```tsx
if (!clusterPreferences[clusterId]) {
  clusterPreferences[clusterId] = {}
}
clusterPreferences.current[clusterId].pinnedResources = pinnnedResources;